### PR TITLE
[cleanup] Remove unused values

### PR DESCRIPTION
**Pull request by adamsmd**
This is a first batch of code cleanups based on dead-code analysis.

They should be checked carefully as some dead code may surprise you.

### CODE REVIEW

File: src/hazelcore/Action_Exp.re [hunk 1]
```diff
@@ -17,30 +17,6 @@ let operator_of_shape =
   | SVBar => None
   };
 
-let shape_of_operator =
```
Comment (cyrus-) [inline src/hazelcore/Action_Exp.re]: leave this one in -- it's a check that new operators have a corresponding shape (leave a comment to this effect)

**Review by cyrus- — CHANGES_REQUESTED** [review]

File: src/hazelcore/Action_Exp.re [hunk 2]
```diff
@@ -72,24 +48,6 @@ let mk_and_ana_fix_ZOpSeq =
   Statics_Exp.ana_fix_holes_z(ctx, u_gen, ([], ExpLineZ(zopseq), []), ty);
 };
 
-/**
```
Comment (cyrus-) [inline src/hazelcore/Action_Exp.re:75]: @dmoon1221 is this function still necessary?

File: src/hazelcore/Action_Pat.re [hunk 3]
```diff
@@ -15,12 +15,6 @@ let operator_of_shape: Action_common.operator_shape => option(UHPat.operator) =
   | SArrow
   | SVBar => None;
 
-let shape_of_operator: UHPat.operator => Action_common.operator_shape =
```
Comment (cyrus-) [inline src/hazelcore/Action_Pat.re:18]: same as above

File: src/hazelcore/Action_Typ.re [hunk 4]
```diff
@@ -17,13 +17,6 @@ let operator_of_shape =
   | SCons => None
   };
 
-let shape_of_operator = (op: UHTyp.operator): Action_common.operator_shape =>
```
Comment (cyrus-) [inline src/hazelcore/Action_Typ.re]: same as above

File: src/hazelcore/CursorPath_common.re [hunk 5]
```diff
@@ -479,48 +459,3 @@ let steps_to_hole = (hole_list: hole_list, u: MetaVar.t): option(steps) =>
...
-
-let rec is_prefix_of = (steps, prefix) =>
-  switch (prefix, steps) {
-  | ([], _) => true
-  | ([_, ..._], []) => false
-  | ([prefix_first, ...prefix_rest], [steps_first, ...steps_rest]) =>
-    prefix_first == steps_first && prefix_rest |> is_prefix_of(steps_rest)
-  };
-
-let rec compare_steps = (steps1, steps2) =>
```
Comment (cyrus-) [inline src/hazelcore/CursorPath_common.re:511]: keep this one

File: src/hazelcore/Deserialize.re [hunk 6]
```diff
@@ -1,10 +0,0 @@
-open Sexplib;
```
Comment (cyrus-) [inline src/hazelcore/Deserialize.re]: keep Serialize and Deserialize

File: src/hazelcore/HTyp.re [hunk 7]
```diff
@@ -108,37 +97,13 @@ let matched_sum =
...
   | _ => None;
 
-let has_matched_list =
-  fun
-  | Hole => true
-  | List(_) => true
-  | _ => false;
-
-/* complete (i.e. does not have any holes) */
-let rec complete =
```
Comment (cyrus-) [inline src/hazelcore/HTyp.re:131]: leave this -- its a terminology thing

File: src/hazelweb/util/JSUtil.re [hunk 8]
```diff
@@ -5,205 +5,15 @@ module Ev = Dom_html.Event;
```
Comment (cyrus-) [inline src/hazelweb/util/JSUtil.re]: leave this unchanged

File: src/hazelcore/Action_Exp.re (hunk 2 above)
Comment (dm0n3y) [reply src/hazelcore/Action_Exp.re:75]: No, its functionality is subsumed by `resurround`

File: src/hazelcore/ZExp.re [hunk 9]
```diff
@@ -483,9 +483,6 @@ let erase_zseq = ZSeq.erase(~erase_zoperand, ~erase_zoperator);
 let mk_ZOpSeq =
   ZOpSeq.mk(~associate=UHExp.associate, ~erase_zoperand, ~erase_zoperator);
 
-let get_err_status = ze => ze |> erase |> UHExp.get_err_status;
-let get_err_status_zblock = zblock =>
-  zblock |> erase_zblock |> UHExp.get_err_status_block;
```
Comment (dm0n3y) [inline src/hazelcore/ZExp.re:488]: Leave these for sake of completeness

File: src/hazelcore/ZTyp.re [hunk 10]
```diff
@@ -48,12 +48,7 @@ and erase_zoperand =
 let mk_ZOpSeq: zseq => zopseq =
   ZOpSeq.mk(~associate=UHTyp.associate, ~erase_zoperand, ~erase_zoperator);
 
-let erase_zseq = zseq =>
-  zseq |> ZSeq.erase(~erase_zoperand, ~erase_zoperator);
-
-let rec is_before = (zty: t): bool => zty |> is_before_zopseq
-and is_before_zopseq = zopseq => ZOpSeq.is_before(~is_before_zoperand, zopseq)
```
Comment (dm0n3y) [inline src/hazelcore/ZTyp.re]: Leave these for completeness

File: src/hazelcore/ZTyp.re [hunk 11]
```diff
@@ -82,10 +71,6 @@ and is_after_zoperand =
   | CursorT(cursor, List(_)) => cursor == OnDelim(1, After)
   | ParenthesizedZ(_) => false
   | ListZ(_) => false;
-let is_after_zoperator: zoperator => bool =
-  fun
-  | (OnOp(After), _) => true
-  | _ => false;
```
Comment (dm0n3y) [inline src/hazelcore/ZTyp.re]: All of the `is_[before|after]_*` functions should remain

**Pull request merged**